### DISCUSSION TITLE
HC_PWN/Asm_x86/

### DISCUSSION THREAD

**Discussion by giscus** [ProbiusOfficial/Hello-CTF#53] [Show and tell]
# HC_PWN/Asm_x86/

开源免费、新手友好的CTF(Capture The Flag,夺旗赛)入门教程

https://ctf.probius.xyz/HC_PWN/Asm_x86/

**Comment by corsnail** [comment]
>可以编译为汇编语言
>```
>.LC0:
>    .string "a == b"
>main:
>    push    rbp
>    mov rbp, rsp
>    mov DWORD PTR -4[rbp], 1
>    mov DWORD PTR -8[rbp], 2
>    mov eax, DWORD PTR -4[rbp]
>    cmp eax, DWORD PTR -8[rbp]
>    jne .L2
>    lea rdi, .LC0[rip]
>    mov eax, 0
>    call    printf@PLT
>.L2:
>    mov eax, 0
>    pop rbp
>    ret
>```
> 可以看到，其实现的原理就是，先将a和b的值分别复制到eax和edx寄存器中，然后比较eax和edx寄存器中的值，如果相等就跳转到.L2这个标签所在的位置，如果不相等就继续往下执行

逻辑控制里的这一段，应该写反了，应该是不相等就跳转到.L2标签，相等就继续往下执行吧。

**Reply by LSQzzx** [reply]
是的
